test/unit/constraints/array.test.ts

```diff
@@ -0,0 +1,113 @@
+import "../metadataShim";
+import {registerJoi, WORKING_SCHEMA_KEY} from "../../../src/core";
+import * as Joi from 'joi';
+import { Max, Length, ArraySchema, Min, Unique } from "../../../src/constraints/array";
+import {Validator} from "../../../src/Validator";
+import { testConstraint } from "../testUtil";
+
+
+registerJoi(Joi);
+
+describe('Array Test', () => {
```

Comment:
:pick: The conventions for the existing constraint tests are for the outer suite to be named like `Array constraints`.